### ISSUE TITLE
Build is failing

### ISSUE THREAD

**Issue by svanteschubert** [dita4publishers/dita4publishers#37]
Hi,

I tried to build dita4publishers with ivy, I usually use Maven but AFAIK Ivy (the Ant-extension) is reusing the Maven repositories.

But already failed getting the dependencies using "ant resolve" from Maven

[ivy:retrieve]          ::::::::::::::::::::::::::::::::::::::::::::::
[ivy:retrieve]          ::          UNRESOLVED DEPENDENCIES         ::
[ivy:retrieve]          ::::::::::::::::::::::::::::::::::::::::::::::
[ivy:retrieve]          :: apache#ant;1.7.1: not found
[ivy:retrieve]          :: apache#commons-logging;1.1: not found
[ivy:retrieve]          :: apache#commons-io;1.4: not found
[ivy:retrieve]          :: apache#commons-cli;1.2: not found
[ivy:retrieve]          :: apache#xml-apis;20080213: not found
[ivy:retrieve]          :: apache#resolver;1.2: not found
[ivy:retrieve]          :: stax#stax-ri;1.0: not found
[ivy:retrieve]          :: saxonica#saxon9;9.1.0.7: not found
[ivy:retrieve]          :: saxonica#saxon9-dom;9.1.0.7: not found
[ivy:retrieve]          :: saxonica#saxon9-xpath;9.1.0.7: not found
[ivy:retrieve]          :: saxonica#saxon9-s9api;9.1.0.7: not found
[ivy:retrieve]          ::::::::::::::::::::::::::::::::::::::::::::::

My guess most of you developers have a running Eclipse environment as the Eclipse .classpath file indicates with all Jars within a /lib directory which does not exist in the git repro.

Would it not be nice if this project could be able to run out-of-the-box with using
https://github.com/dita4publishers/dita4publishers/blob/develop/ivy.xml

The usual problem was that the artifact prefix was incorrect.
For instance: 
[ivy:retrieve]          :: apache#commons-cli;1.2: not found
If searched at the Maven repro:
https://mvnrepository.com/artifact/commons-cli/commons-cli
Would be in ivy.xml not
<dependency org="apache" name="commons-cli" rev="1.2"/>
but 
<dependency org="commons-cli" name="commons-cli" rev="1.2"/>

As the libraries were quite old, I felt bold and tried to update to the latest version of JARs to see if the sources are still building, but how are the sources being build? ;-)

Especially when looking the Saxon9-HE (home edition - opensource)
http://www.saxonica.com/products/feature-matrix-9-9.xml
seemed to be worth a try to go on the latest work of technology (as well to be able to report issues).

Unfortunately, I realized today that their Maven artefact was broken, 
https://saxonica.plan.io/issues/4158
but locally I could fix everything for me and it is now reported.

Still I would love to test the latest JAR on the source to build..
Please find the updated ivy.xml as an experimental draft in the very end of this issue (you need to do the saxon fix of my issue above to make it work)

PS: The Travis builds are not working see
https://travis-ci.org/dita4publishers/dita4publishers/jobs/451400857
as 
https://github.com/dita4publishers/dita4publishers/edit/develop/.travis.yml
is using   
- oraclejdk7 
while there is only Java 8 on the machine, you might want to try a
 - oraclejdk8

------------- draft of version update of ivy.xml -------------

<ivy-module version="2.0">
	<info organisation="dita4publishers" module="dita4publishers"/>
	<dependencies>
	    
		<dependency org="org.apache.ant" name="ant" rev="1.10.5"/>
		
		<dependency org="commons-logging" name="commons-logging" rev="1.2"/>
		
		<dependency org="commons-io" name="commons-io" rev="2.6"/>
		
		<dependency org="commons-cli" name="commons-cli" rev="1.4"/>
		
		<dependency org="org.apache.logging.log4j" name="log4j-core" rev="2.11.2" transitive="false"/>
		
		
		
		
	
		
		

    	
		<dependency org="net.sf.saxon" name="Saxon-HE" rev="9.9.1-1"/>
		
		<dependency org="xom" name="xom" rev="1.2.10"/>

		
		<conflict manager="all"/>
	</dependencies>
</ivy-module>

**Comment by drmacro** [comment]
I haven't done anything with the D4P Java code in literally years. It would certainly be appropriate to update the project to use Maven--all the dependencies should be there.

Likewise, there is probably code that depends on the old Saxon 9.1 API. 

I've recently updated the DITA Community i18n library to work with Saxon 9.8 so I know how to do the updating (and how to make it work with OT 3.3+.

Most of D4P is just XSLT. Which part of the Java code are you interested in? A lot of it is really old and probably of dubious value at this point.

**Comment by svanteschubert** [comment]
Hello Eliot,

Thanks for the quick response, I just worked myself into the project as it seemed quite appropriate for a customer in-spe here in Berlin. Some publisher, who wanted to do DITA to Indesigh automation, but I am afraid he decided to solve the problem in-house and I could not even get to know the name of the company to make some promotion for your project. I hope the headhunter has forwarded my notes, in this case there is some chance that someone gets in contact with you.

In other words, I have to spend my time with other projects..

All the best!
Svante
